src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -477,7 +477,9 @@ private boolean canDoExactSearch(final int filterIdsCount) {
             return filterThresholdValue >= filterIdsCount;
         }
         // if no setting is set, then use the default max distance computation value to see if we can do exact search.
-        return KNNConstants.MAX_DISTANCE_COMPUTATIONS >= filterIdsCount * knnQuery.getQueryVector().length;
+        return KNNConstants.MAX_DISTANCE_COMPUTATIONS >= filterIdsCount * (knnQuery.getVectorDataType() == VectorDataType.FLOAT
+            ? knnQuery.getQueryVector().length
+            : knnQuery.getByteQueryVector().length);
```

Comment:
with hamming distance we can do more computation for exact search. I would recommend adding a TODO here to ensure that we revisit the number of distance computation for different fields.

Reply:
ack